volgen/core.go

```diff
@@ -71,32 +73,43 @@ func GenerateVolfile(vinfo *volume.Volinfo) error {
 	return nil
 }
 
-// Fix this: Move it to utils and make it return string
-func getClientFilePath(vinfo *volume.Volinfo, path *string) error {
-	vdir := utils.GetVolumeDir(vinfo.Name)
-
-	// Create volume directory (/var/lib/glusterd/vols/<VOLNAME>)
-	err := os.MkdirAll(vdir, os.ModeDir|os.ModePerm)
-	if err != nil {
-		return err
-	}
-
-	*path = fmt.Sprintf("%s/trusted-%s.tcp-fuse.vol", vdir, vinfo.Name)
-	return err
+func getClientVolFilePath(vinfo *volume.Volinfo) string {
+	return fmt.Sprintf("%s/trusted-%s.tcp-fuse.vol", utils.GetVolumeDir(vinfo.Name), vinfo.Name)
```

Comment:
When generating paths, it's better to use the `path` package. Generate the filename, and the generate the full path by doing a `path.Join(dir,filename)`.